src/native/managed/cdac/mscordaccore_universal/Legacy/SOSDacImpl.cs

```diff
@@ -692,28 +680,25 @@ int ISOSDacInterface.GetJumpThunkTarget(void* ctx, ClrDataAddress* targetIP, Clr
         try
         {
             // API is implemented for x64 only
-            if (_target.Contracts.RuntimeInfo.GetTargetArchitecture() == RuntimeInfoArchitecture.X64)
-            {
-                IPlatformAgnosticContext context = IPlatformAgnosticContext.GetContextForPlatform(_target);
+            if (_target.Contracts.RuntimeInfo.GetTargetArchitecture() != RuntimeInfoArchitecture.X64)
+                throw new ReturnHResultException(HResults.E_FAIL);
```

Comment:
`Marshal.ThrowExceptionForHR(HResults.E_FAIL)` will give you an exception with the right HResult here.